include/p4mlir/Dialect/P4HIR/P4HIR_TypeInterfaces.td

```diff
@@ -176,4 +175,35 @@ def HasDefaultValue : TypeInterface<"HasDefaultValue"> {
   let cppNamespace = "::P4::P4MLIR::P4HIR";
 }
 
+def EnumRepresentationInterface : TypeInterface<"EnumRepresentationInterface"> {
+  let description = [{
+    Interface to specify the underlying representation of an enum type when it is
+    lowered to serialized form. This allows downstream users to provide a policy
+    for the bit-width of enums.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/        "Get the underlying type for an enum type",
+      /*returnType=*/  "mlir::Type",
+      /*methodName=*/  "getUnderlyingType",
+      /*args=*/        (ins)
+    >,
+    InterfaceMethod<
+      /*desc=*/        "Whether this enum should be converted to serializable form",
+      /*returnType=*/  "bool",
+      /*methodName=*/  "shouldConvert",
+      /*args=*/        (ins)
+    >,
+    InterfaceMethod<
+      /*desc=*/        "Get the numeric encoding for a specific enum field",
```

Comment:
I am wondering, do we have any restrictions on `getUnderlyingType`? Can it return, say `p4hir.string`? Given the current interface – yes. Then `getEncoding` likely should be changed? Looks like the intention is that `getUnderlyingType` would always return `BitsType`, no? The it should be explicitly specified this way